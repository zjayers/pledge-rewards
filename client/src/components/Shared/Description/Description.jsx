import React from 'react';
import { DescCollapseBtn, DescriptionContainer } from './Description.style';

const Description = (props) => {
  const { description, descOpen, activated, setDescOpen } = props;
  let { descClicked } = props;

  if (!descClicked) {
    descClicked = () => setDescOpen(!descOpen);
  }

  /**
   * Render Description Button
   * Renders the 'Read more...' or 'Less' button depending on if the description box is open or collapsed
   * @returns {string|*}
   */
  const renderDescButton = () => {
    if (activated) {
      return descOpen ? 'Less' : 'Read more...';
    }

    /* Return a Fragment */
    return <></>;
  };

  return (
    <DescriptionContainer open={descOpen} activated={activated}>
      <p>{description}</p>
      <DescCollapseBtn onClick={descClicked} open={descOpen}>
        {renderDescButton()}
      </DescCollapseBtn>
    </DescriptionContainer>
  );
};

export default Description;
