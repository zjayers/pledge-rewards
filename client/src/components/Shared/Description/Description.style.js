import styled from 'styled-components';

export const DescriptionContainer = styled.div`
  font-size: 1.2rem;
  line-height: 1.5;
  position: relative;
  height: ${(props) => {
    if (props.activated) {
      return props.open ? 'auto' : '40px';
    }

    return 'auto';
  }};
  color: ${(props) => props.theme.colors.flavorText};
  display: grid;
  grid-template-rows: ${(props) => (props.activated ? '1fr 20px;' : '1fr')};
  overflow: hidden;
  margin-bottom: ${(props) => (props.open ? '1rem' : '20px')};
`;

export const DescCollapseBtn = styled.p`
  color: ${(props) => props.theme.colors.text};
  background: linear-gradient(
    to right,
    rgba(255, 255, 255, 0) 0%,
    rgba(255, 255, 255, 1) 20%
  );
  width: 80px;
  text-align: end;
  cursor: pointer;
  position: absolute;
  right: 5px;
  bottom: 0;
`;
