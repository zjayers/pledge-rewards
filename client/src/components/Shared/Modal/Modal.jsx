import React from 'react';
import ReactDOM from 'react-dom';
import { Fader, InfoCard } from './Modal.style';
import styles from './Modal.module.scss';

import Heading from '../Heading/Heading';
import SubHeading from '../SubHeading/SubHeading';

const Modal = ({ setModalOpen, project }) => {
  const { creator, location } = project;

  const renderModal = () => {
    const toggleModal = (e) => {
      e.stopPropagation();
      setModalOpen(false);
    };

    return (
      <Fader onClick={toggleModal}>
        <InfoCard onClick={(e) => e.stopPropagation()}>
          <p>About the creator</p>
          <Heading heavy>{creator}</Heading>
          <SubHeading>{location}</SubHeading>
          <p>
            Lorem ipsum dolor sit amet, consectetur adipisicing elit. Adipisci
            consequatur dicta, eveniet exercitationem laborum minus mollitia
            neque odio perferendis, placeat quod rem ut voluptates. Consequuntur
            est in iure minima perspiciatis? Lorem ipsum dolor sit amet,
            consectetur adipisicing elit. Ad cumque doloremque maxime, nihil non
            omnis sapiente. Commodi inventore molestias ratione reiciendis
            sequi. Asperiores beatae eos id nam quam quidem vero! Lorem ipsum
            dolor sit amet, consectetur adipisicing elit. Ad corporis deleniti,
            distinctio esse, labore modi odit officiis pariatur provident qui
            rerum veniam veritatis! Ad doloremque fuga harum suscipit vitae.
            Modi! Lorem ipsum dolor sit amet, consectetur adipisicing elit.
            Corporis cupiditate temporibus voluptate! Aliquid deserunt dicta
            expedita hic libero nihil qui sed. Cumque debitis dolorem earum enim
            ipsa modi tempora ut!
          </p>
          <div className={styles.flex}>
            <div className={styles.flexColumn}>
              <ul>
                <li>
                  <i className='fas fa-check' style={{ color: '#009E74' }} />
                </li>
                <li>
                  <i className='fas fa-lock' />
                </li>
                <li>
                  <i className='fab fa-facebook' style={{ color: '#39579A' }} />
                </li>
                <li>
                  <i
                    className='fab fa-kickstarter-k'
                    style={{ color: '#057362' }}
                  />
                </li>
              </ul>
            </div>
            <div className={styles.flexColumn}>
              <ul>
                <li>{creator}</li>
                <li>Last login</li>
                <li>Not connected</li>
                <li style={{ color: '#27A47E' }}>1 created Â· 0 backed</li>
              </ul>
            </div>
          </div>
        </InfoCard>
        <span>
          {/* eslint-disable-next-line jsx-a11y/click-events-have-key-events,jsx-a11y/no-static-element-interactions */}
          <u onClick={toggleModal}>Close</u>
        </span>
      </Fader>
    );
  };

  return ReactDOM.createPortal(
    renderModal(),
    document.getElementById('modal-root')
  );
};

export default Modal;
