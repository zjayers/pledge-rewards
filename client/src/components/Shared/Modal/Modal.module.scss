.flex {
  display: flex;
}

.flexColumn {
  display: flex;
  flex-direction: column;

  & ul {
    list-style: none;
  }

  & li {
    font-size: 12px;
    margin: 12px 10px 12px 0;
  }
}
