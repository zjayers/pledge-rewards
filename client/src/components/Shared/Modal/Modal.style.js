import styled from 'styled-components';

export const Fader = styled.div`
  position: absolute;
  width: 100%;
  height: 100%;
  background: rgba(255, 255, 255, 0.9);
  display: flex;
  flex-direction: column;

  & span {
    display: inline;
    font-size: 12px;
    position: absolute;
    left: 0;
    right: 0;
    bottom: 12vh;
    text-align: center;
  }

  & span u {
    cursor: pointer;
    color: #656969;
    &:hover {
      color: #28282b;
    }
  }
`;

export const InfoCard = styled.div`
  width: 40vw;
  max-height: 70vh;
  box-shadow: 0 0 4px 0 rgba(168, 167, 164, 0.4);
  padding: 2.4rem;
  overflow-y: auto;
  color: #282828;
  z-index: 999;
  left: 0;
  right: 0;
  top: 0;
  margin: auto;

  & p:first-of-type {
    font-size: 18px;
    text-align: center;
    margin: 0 auto 60px;
  }

  & h2 {
    font-size: 2.4rem;
  }

  & p:last-of-type {
    margin-top: 10px;
    font-size: 14px;
    line-height: 1.4;
    margin-bottom: 30px;
  }
`;
