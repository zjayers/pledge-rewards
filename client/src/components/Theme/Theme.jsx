import React from 'react';
import { ThemeProvider } from 'styled-components';

const theme = {
  colors: {
    borderGrey: '#9B9E9E',
    highlightGreen: '#009E74',
    buttonGreen: '#009E74',
    gradientBlue: '#DBE7FE',
    gradientOrange: '#FFF2EC',
    text: '#282828',
    flavorText: '#888B8B',
    linkText: '#498F80',
  },
  fonts: ['Muli', 'sans-serif'],
  fontSizes: {
    small: '1rem',
    medium: '1.8rem',
    large: '3rem',
  },
};

const Theme = ({ children }) => (
  <ThemeProvider theme={theme}>{children}</ThemeProvider>
);

export default Theme;
