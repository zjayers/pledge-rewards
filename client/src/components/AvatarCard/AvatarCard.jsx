/* Import Modules */
import React, { useEffect, useState } from 'react';
/* Import Components */
import Container from '../Shared/Container/Container';
import Heading from '../Shared/Heading/Heading';
import Avatar from '../Shared/Avatar/Avatar';
import { getUserInfo } from '../../services/apiService';
import Description from '../Shared/Description/Description';
import Modal from '../Shared/Modal/Modal';

/**
 * Avatar Card Component
 * @returns {*}
 * @constructor
 */
const AvatarCard = () => {
  const [project, setProject] = useState({
    creator: '',
    rewards: [{ description: '' }],
  });
  const [modalOpen, setModalOpen] = useState(false);

  useEffect(() => {
    getUserInfo().then((response) => {
      if (response.data.length > 0) {
        setProject(response.data[0]);
      }
    });
  }, []);

  /**
   * Format Username
   * Remove underscore from username
   * @returns {string}
   */
  const formatUsername = () => {
    project.creator = project.creator.replace('_', ' ').replace('.', ' ');
    return project.creator;
  };

  /* Return the JSX to render */
  return (
    <Container activated padding='5.2rem'>
      <Avatar />
      <Heading heavy>{formatUsername()}</Heading>
      <p>1 created Â· 0 backed</p>
      <Description
        descOpen={false}
        activated
        setDescOpen={setModalOpen}
        description={project.rewards[0].description}
      />
      {modalOpen ? (
        <Modal
          modalOpen={modalOpen}
          setModalOpen={setModalOpen}
          project={project}
        />
      ) : (
        <> </>
      )}
    </Container>
  );
};

/* Export the component */
export default AvatarCard;
