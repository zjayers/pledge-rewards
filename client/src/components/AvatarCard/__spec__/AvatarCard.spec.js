import React from 'react';
import { mount } from 'enzyme';
import { act } from 'react-dom/test-utils';
import toJson from 'enzyme-to-json';
import AvatarCard from '../AvatarCard';
import Theme from '../../Theme/Theme';
import Description from '../../Shared/Description/Description';
import { DescCollapseBtn } from '../../Shared/Description/Description.style';
import Modal from '../../Shared/Modal/Modal';

const axios = require('axios');
const MockAdapter = require('axios-mock-adapter');

describe('AvatarCard Component', () => {
  const mockReward = {
    title: 'Handcrafted Metal Bacon',
    pledgeAmount: 535,
    description:
      'Dolorem nulla distinctio temporibus fuga quibusdam qui qui cum sint. Consequatur quia corporis optio non inventore ab qui labore. Quam necessitatibus iure qui. Fuga laborum error. Optio minima quibusdam optio eum.',
    deliveryMonth: 'November',
    deliveryYear: 2020,
    shippingType: 'vertical',
    rewardQuantity: 55709,
    timeLimit: 69840,
    projectId: 1,
    rewardItems: 'Bike,Table,Shirt,Hat,Cheese',
  };

  const mockProject = {
    title: 'Fantastic Granite Table',
    creator: 'Bob',
    subtitle: 'Fundamental incremental extranet',
    category: 'Games',
    subcategory: 'Handcrafted',
    location: 'West Prudence, CA',
    heroImage: 'http://lorempixel.com/640/480/nature',
    heroVideo: 'https://ytroulette.com/',
    launchDate: '2021-03-15T21:00:17.200Z',
    campaignDuration: 132,
    budget: 530,
    fundingGoal: 564,
    rewards: [mockReward],
  };

  let component;

  const mock = new MockAdapter(axios);
  mock.onGet(/.*/).reply(200, [mockProject]);

  beforeEach(async () => {
    component = mount(
      <Theme>
        <AvatarCard />
      </Theme>
    );

    await act(async () => {
      await new Promise((resolve) => setImmediate(resolve));
      component.update();
    });
  });

  afterEach((done) => {
    component.unmount();
    done();
  });

  it('should render without failure', () => {
    expect(component).toBeDefined();
  });

  it('should not show the Modal by default', () => {
    expect(component.find(Modal).length).toEqual(0);
  });

  it('should show the modal when description button is clicked', () => {
    // add a div with #modal-root id to the global body
    const modalRoot = global.document.createElement('div');
    modalRoot.setAttribute('id', 'modal-root');
    const body = global.document.querySelector('body');
    body.appendChild(modalRoot);

    const desc = component.find(Description);
    expect(desc.length).toEqual(1);
    const descButton = desc.find(DescCollapseBtn);
    descButton.simulate('click', () => {});
    expect(component.find(Modal).length).toEqual(1);
  });

  it('should match the test snapshot', () => {
    expect(toJson(component)).toMatchSnapshot();
  });
});
