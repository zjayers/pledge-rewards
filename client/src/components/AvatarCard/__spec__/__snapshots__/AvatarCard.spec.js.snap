// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AvatarCard Component should match the test snapshot 1`] = `
<Theme>
  <ThemeProvider
    theme={
      Object {
        "colors": Object {
          "borderGrey": "#9B9E9E",
          "buttonGreen": "#009E74",
          "flavorText": "#888B8B",
          "gradientBlue": "#DBE7FE",
          "gradientOrange": "#FFF2EC",
          "highlightGreen": "#009E74",
          "linkText": "#498F80",
          "text": "#282828",
        },
        "fontSizes": Object {
          "large": "3rem",
          "medium": "1.8rem",
          "small": "1rem",
        },
        "fonts": Array [
          "Muli",
          "sans-serif",
        ],
      }
    }
  >
    <AvatarCard>
      <_default
        activated={true}
        padding="5.2rem"
      >
        <Containerstyle___default
          activated={true}
          padding="5.2rem"
        >
          <div
            className="Containerstyle___default-sc-1q8wjhx-0 kSgUlC"
          >
            <_default>
              <Avatarstyle___default>
                <div
                  className="Avatarstyle___default-s7ub6l-0 iayjJg"
                >
                  [object Object]
                </div>
              </Avatarstyle___default>
            </_default>
            <_default
              heavy={true}
            >
              <Headingstyle___default
                heavy={true}
              >
                <h2
                  className="Headingstyle___default-dlpxpq-0 iWDnYS"
                >
                  Bob
                </h2>
              </Headingstyle___default>
            </_default>
            <p>
              1 created Â· 0 backed
            </p>
            <Description
              activated={true}
              descOpen={false}
              description="Dolorem nulla distinctio temporibus fuga quibusdam qui qui cum sint. Consequatur quia corporis optio non inventore ab qui labore. Quam necessitatibus iure qui. Fuga laborum error. Optio minima quibusdam optio eum."
              setDescOpen={[Function]}
            >
              <Descriptionstyle__DescriptionContainer
                activated={true}
                open={false}
              >
                <div
                  className="Descriptionstyle__DescriptionContainer-wtqy6i-0 hRNJRj"
                  open={false}
                >
                  <p>
                    Dolorem nulla distinctio temporibus fuga quibusdam qui qui cum sint. Consequatur quia corporis optio non inventore ab qui labore. Quam necessitatibus iure qui. Fuga laborum error. Optio minima quibusdam optio eum.
                  </p>
                  <Descriptionstyle__DescCollapseBtn
                    onClick={[Function]}
                    open={false}
                  >
                    <p
                      className="Descriptionstyle__DescCollapseBtn-wtqy6i-1 loBJcu"
                      onClick={[Function]}
                      open={false}
                    >
                      Read more...
                    </p>
                  </Descriptionstyle__DescCollapseBtn>
                </div>
              </Descriptionstyle__DescriptionContainer>
            </Description>
             
          </div>
        </Containerstyle___default>
      </_default>
    </AvatarCard>
  </ThemeProvider>
</Theme>
`;
